deploy/lib/server_config.rb

```diff
@@ -1026,8 +1032,22 @@ def save_files_to_fs(target_db, target_dir)
           # create the directory so that it will exist when we try to save files
           Dir.mkdir("#{target_dir}" + uri)
         else
-          r = go("#{@protocol}://#{@hostname}:#{@bootstrap_port}/qconsole/endpoints/view.xqy?dbid=#{db_id}&uri=#{uri}", "get")
-          File.open("#{target_dir}#{uri}", 'wb') { |file| file.write(r.body) }
+          # check type, if REST use that endpoint, otherwise use a query
+          if (port != nil)
+            r = go("#{@protocol}://#{@hostname}:#{@bootstrap_port}/qconsole/endpoints/view.xqy?dbid=#{db_id}&uri=#{uri}", "get")
+            file_content = r.body
+          else
+            # Note : There are limitations to doing it this way for non REST DBs; only text files are captured (xqy, html, etc..) 
+            # other types such as images or java scripts will not be grabbed. Also this method is slower than REST or XCC.
```

Comment:
The problem in #163 was that the REST API (at least at the time) was returning a JSON description of the binary files, instead of the file itself.